stream/statelib/src/test/java/org/apache/bookkeeper/statelib/impl/kv/TestRocksdbKVStoreCheckpoint.java

```diff
@@ -0,0 +1,97 @@
+/*
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.apache.bookkeeper.statelib.impl.kv;
+
+import static org.junit.Assert.assertEquals;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.bookkeeper.statelib.impl.rocksdb.checkpoint.CheckpointInfo;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.rules.TestName;
+
+
+
+
+/**
+ * Test cases for Rocksdb KV Store with checkpoints.
+ */
+@Slf4j
+public class TestRocksdbKVStoreCheckpoint {
```

Comment:
following up on 

> So if we restore from older checkpoint, we will replay from older TxId. This can also detect that there is missing data if the starting TxId is not available in the Journal. I am not sure how that is handled currently by DistributedLog. Ideally that should also throw an exception during journal replay.

This change needs a test reproing this case to confirm that the change does not introduce a silent data loss.

Reply:
DistributedLog starts the replay after the specified transactions. So it skips over the missing records. To address this I have added an explicit verification for starting transaction at the start of journal replay. Also added a test-case to test the same (see testMissingData() above).